### PR TITLE
use new tree parallel transformer and add performance test

### PR DESCRIPTION
**Pull request by andimarek**
this improves the performance more by using the new tree parallel transformer

### CODE REVIEW

File: src/main/java/graphql/nadel/engine/StrategyUtil.java [hunk 1]
```diff
@@ -1,47 +1,67 @@
...
+        Comparator<NodeZipper<ExecutionResultNode>> comparator = (node1, node2) -> {
+            if (node1 == node2) {
+                return 0;
+            }
+            List<Breadcrumb<ExecutionResultNode>> breadcrumbs1 = node1.getBreadcrumbs();
+            List<Breadcrumb<ExecutionResultNode>> breadcrumbs2 = node2.getBreadcrumbs();
+            if (breadcrumbs1.size() != breadcrumbs2.size()) {
+                return Integer.compare(breadcrumbs1.size(), breadcrumbs2.size());
+            }
+            for (int i = breadcrumbs1.size() - 1; i >= 0; i--) {
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/engine/StrategyUtil.java]: Put Comparable on Breadcrumbs?

File: src/main/java/graphql/nadel/engine/StrategyUtil.java [hunk 2]
```diff
@@ -64,21 +84,15 @@ public static ExecutionStepInfo createRootExecutionStepInfo(GraphQLSchema graphQ
...
-    public static <T extends ExecutionResultNode> T changeEsiInResultNode(T executionResultNode, ExecutionStepInfo newEsi) {
+    public static <T extends ExecutionResultNode> T
+    changeEsiInResultNode(T executionResultNode, ExecutionStepInfo newEsi) {
         return (T) executionResultNode.withNewExecutionStepInfo(newEsi);
     }
 
-    public static <T extends ExecutionResultNode> T changeFieldInResultNode(T executionResultNode, MergedField mergedField) {
+    public static <T extends ExecutionResultNode> T
+    changeFieldInResultNode(T executionResultNode, MergedField mergedField) {
         ExecutionStepInfo newStepInfo = executionResultNode.getExecutionStepInfo().transform(builder -> builder.field(mergedField));
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/engine/StrategyUtil.java]: weird indenting??

File: src/main/java/graphql/nadel/engine/StrategyUtil.java [hunk 3]
```diff
@@ -64,21 +84,15 @@ public static ExecutionStepInfo createRootExecutionStepInfo(GraphQLSchema graphQ
...
         return (T) executionResultNode.withNewExecutionStepInfo(newStepInfo);
     }
 
-
-    public static List<HydrationTransformation> getHydrationTransformations(Collection<FieldTransformation> transformations) {
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/engine/StrategyUtil.java]: not used anymore?

Comment (andimarek) [reply src/main/java/graphql/nadel/engine/StrategyUtil.java]: not used, yes

**Pull request merged**